analysisservice/src/main/java/nu/yona/server/analysis/service/AnalysisEngineService.java

```diff
@@ -121,61 +123,59 @@ private ZonedDateTime correctTime(Duration deviceTimeOffset, ZonedDateTime time)
 		return time.minus(deviceTimeOffset);
 	}
 
-	private void analyze(ActivityPayload payload, UserAnonymizedDTO userAnonymized,
-			Set<ActivityCategoryDTO> matchingActivityCategories)
+	private void analyze(ActivityPayload payload, Set<ActivityCategoryDTO> matchingActivityCategories)
 	{
-		Set<GoalDTO> matchingGoalsOfUser = determineMatchingGoalsForUser(userAnonymized, matchingActivityCategories,
+		Set<GoalDTO> matchingGoalsOfUser = determineMatchingGoalsForUser(payload.userAnonymized, matchingActivityCategories,
 				payload.startTime);
 		for (GoalDTO matchingGoalOfUser : matchingGoalsOfUser)
 		{
-			addOrUpdateActivity(payload, userAnonymized, matchingGoalOfUser);
+			addOrUpdateActivity(payload, matchingGoalOfUser);
 		}
 	}
 
-	private void addOrUpdateActivity(ActivityPayload payload, UserAnonymizedDTO userAnonymized, GoalDTO matchingGoal)
+	private void addOrUpdateActivity(ActivityPayload payload, GoalDTO matchingGoal)
 	{
-		if (isCrossDayActivity(payload, userAnonymized))
+		if (isCrossDayActivity(payload))
 		{
 			// assumption: activity never crosses 2 days
 			ActivityPayload truncatedPayload = ActivityPayload.copyTillEndTime(payload,
-					getEndOfDay(payload.startTime, userAnonymized));
+					getEndOfDay(payload.startTime, payload.userAnonymized));
 			ActivityPayload nextDayPayload = ActivityPayload.copyFromStartTime(payload,
-					getStartOfDay(payload.endTime, userAnonymized));
+					getStartOfDay(payload.endTime, payload.userAnonymized));
 
-			addOrUpdateDayTruncatedActivity(truncatedPayload, userAnonymized, matchingGoal);
-			addOrUpdateDayTruncatedActivity(nextDayPayload, userAnonymized, matchingGoal);
+			addOrUpdateDayTruncatedActivity(truncatedPayload, matchingGoal);
+			addOrUpdateDayTruncatedActivity(nextDayPayload, matchingGoal);
 		}
 		else
 		{
-			addOrUpdateDayTruncatedActivity(payload, userAnonymized, matchingGoal);
+			addOrUpdateDayTruncatedActivity(payload, matchingGoal);
 		}
 	}
 
-	private void addOrUpdateDayTruncatedActivity(ActivityPayload payload, UserAnonymizedDTO userAnonymized, GoalDTO matchingGoal)
+	private void addOrUpdateDayTruncatedActivity(ActivityPayload payload, GoalDTO matchingGoal)
 	{
-		DayActivityCacheResult dayActivity = getRegisteredDayActivity(payload, userAnonymized, matchingGoal);
-		Activity lastRegisteredActivity = dayActivity.content == null ? null : dayActivity.content.getLastActivity();
-		if (canCombineWithLastRegisteredActivity(payload, lastRegisteredActivity))
+		ActivityCacheResult activity = getRegisteredDayActivity(payload, matchingGoal);
```

Comment:
Can you rename ``activity`` into ``activityCacheResult``? It confused me many times before that it's not an activity but a cache result.

Please do that in other methods too.